agent/http/server.go

```diff
@@ -62,12 +66,13 @@ type Server struct {
 }
 
 // NewHTTPServer creates a new agent HTTP server.
-func NewHTTPServer(debug bool, cfg *config.HTTP, log hclog.Logger, agent AgentHTTP) (*Server, error) {
+func NewHTTPServer(debug, prom bool, cfg *config.HTTP, log hclog.Logger, agent AgentHTTP) (*Server, error) {
```

Comment:
I think we're getting to the point where the input to `NewHTTPServer` would be better as a struct 😄

Reply:
Agreed and this did cross my mind that we just pass the whole config object. I'll follow this up in a new PR to refactor the func signature.

Reply:
I think passing the whole config would be too much, just the current params in a `HTTPServerConfig` struct should be enough.